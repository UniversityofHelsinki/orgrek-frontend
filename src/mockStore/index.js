export * from './mockApi';
export * from './mockDay';
export * from './mockEditMode';
export * from './mockHierarchy';
export * from './mockHierarchyFilters';
export * from './mockNode';
export * from './mockNodeDetailsView';
export * from './mockStore';
export * from './mockTree';
export * from './mockUser';
