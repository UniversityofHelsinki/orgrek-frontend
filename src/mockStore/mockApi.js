import { rest } from 'msw';
import format from 'date-fns/format';

const formatApiDate = (date) => format(date, 'd.M.yyyy');

const baseUrl = `${process.env.REACT_APP_ORGREK_BACKEND_SERVER || ''}/api`;

export const mockGetNode = (nodeId, body) =>
  rest.get(`${baseUrl}/node/${nodeId}`, (req, res, ctx) => res(ctx.json(body)));

export const mockGetAttributes = (nodeId, body, date) => {
  return rest.get(
    `${baseUrl}/node/${nodeId}/virallinen/${formatApiDate(
      date || new Date()
    )}/attributes`,
    (req, res, ctx) => res(ctx.json(body))
  );
};

export const mockPutNameAttributes = (nodeId) =>
  rest.put(`${baseUrl}/node/${nodeId}/attributes/names`, (req, res, ctx) =>
    res(ctx.delay(2000), ctx.json(req.json()))
  );

export const mockGetTypeAttributes = (nodeId, body) =>
  rest.get(`${baseUrl}/node/${nodeId}/attributes/types`, (req, res, ctx) =>
    res(ctx.json(body))
  );

export const mockPutTypeAttributes = (nodeId) =>
  rest.put(`${baseUrl}/node/${nodeId}/attributes/types`, (req, res, ctx) =>
    res(ctx.delay(2000), ctx.json(req.json()))
  );

export const mockGetValidHierarchyFilters = (date, body) =>
  rest.get(
    `${baseUrl}/hierarchyFilters/${formatApiDate(date)}`,
    (req, res, ctx) => res(ctx.json(body))
  );

export const mockGetCodeAttributes = (nodeId, body, date) =>
  rest.get(`${baseUrl}/node/${nodeId}/attributes/codes`, (req, res, ctx) =>
    res(ctx.json(body))
  );

export const mockSaveCodeAttributes = (nodeId) =>
  rest.put(`${baseUrl}/node/${nodeId}/attributes/codes`, (req, res, ctx) =>
    res(ctx.delay(2000), ctx.json(req.json()))
  );

export const mockGetOtherAttributes = (nodeId, selectedHierarchies, body) =>
  rest.get(
    `${baseUrl}/node/${nodeId}/attributes/others/hierarchies/${selectedHierarchies}`,
    (req, res, ctx) => res(ctx.json(body))
  );

export const mockSaveOtherAttributes = (nodeId) =>
  rest.put(`${baseUrl}/node/${nodeId}/attributes/others`, (req, res, ctx) =>
    res(ctx.delay(2000), ctx.json(req.json()))
  );

export const mockGetAttributeKeys = ({ selectedHierarchies, sections }, body) =>
  rest.get(
    `${baseUrl}/hierarchyFilters/${selectedHierarchies}/${
      (sections && sections.toString()) || 'virallinen'
    }/attributes/keys`,
    (req, res, ctx) => res(ctx.json(body))
  );

export const mockGetFullNames = (nodeId, date, body) =>
  rest.get(
    `${baseUrl}/node/fullname/${nodeId}/${formatApiDate(date)}`,
    (req, res, ctx) => res(ctx.json(body))
  );

export const mockGetAllFullNames = (nodeId, body) =>
  rest.get(`${baseUrl}/node/fullname/all/${nodeId}`, (req, res, ctx) =>
    res(ctx.json(body))
  );

export const mockGetFavorableFullNames = (nodeId, date, body) =>
  rest.get(
    `${baseUrl}/node/fullname/favorable/${nodeId}/${formatApiDate(date)}`,
    (req, res, ctx) => res(ctx.json(body))
  );

export const mockGetParents = (nodeId, date, hierarchies, body) =>
  rest.get(
    `${baseUrl}/node/${nodeId}/${date.toLocaleDateString(
      'fi-FI'
    )}/parents/${hierarchies}`,
    (req, res, ctx) => res(ctx.json(body))
  );

export const mockSaveParents = () =>
  rest.put(`${baseUrl}/node/parents/update`, (req, res, ctx) =>
    res(ctx.delay(2000), ctx.json(req.json()))
  );

export const mockGetChildren = (nodeId, date, hierarchies, body) =>
  rest.get(
    `${baseUrl}/node/${nodeId}/${date.toLocaleDateString(
      'fi-FI'
    )}/children/${hierarchies}`,
    (req, res, ctx) => res(ctx.json(body))
  );

export const mockSaveChildren = () =>
  rest.put(`${baseUrl}/node/children/update`, (req, res, ctx) =>
    res(ctx.delay(2000), ctx.json(req.json()))
  );

export const mockGetPredecessors = (nodeId, date, body) =>
  rest.get(
    `${baseUrl}/node/${nodeId}/${date.toLocaleDateString(
      'fi-FI'
    )}/predecessors`,
    (req, res, ctx) => res(ctx.json(body))
  );

export const mockGetSuccessors = (nodeId, date, body) =>
  rest.get(
    `${baseUrl}/node/${nodeId}/${date.toLocaleDateString('fi-FI')}/successors`,
    (req, res, ctx) => res(ctx.json(body))
  );

export const mockSaveSuccessors = (nodeId) =>
  rest.put(`${baseUrl}/node/${nodeId}/successors`, (req, res, ctx) =>
    res(ctx.delay(2000), ctx.json(req.json()))
  );

export const mockGetSectionTypeAttributes = (sectionType, body) =>
  rest.get(
    `${baseUrl}/node/section/${sectionType}/attributes`,
    (req, res, ctx) => res(ctx.json(body))
  );

export const mockGetTree = ({ hierarchies, selectedDay }, body) =>
  rest.get(
    `${baseUrl}/tree/${hierarchies}/${selectedDay.toLocaleDateString('FI-fi')}`,
    (req, res, ctx) => res(ctx.json(body))
  );

export const mockGetNodeValidity = (id, body) =>
  rest.get(`${baseUrl}/node/${id}`, (req, res, ctx) => res(ctx.json(body)));

export const mockSaveNodeValidity = (id) =>
  rest.put(`${baseUrl}/node/${id}/update`, (req, res, ctx) =>
    res(ctx.delay(2000), ctx.json(req.json()))
  );

/**
 * @deprecated not needed after everything has been migrated to RTK Query
 */
export const mockGetParentsDeprecated = (nodeId, date, hierarchies, body) =>
  rest.get(
    `${baseUrl}/node/parents/${nodeId}/${formatApiDate(date)}/${hierarchies}`,
    (req, res, ctx) => res(ctx.json(body))
  );
