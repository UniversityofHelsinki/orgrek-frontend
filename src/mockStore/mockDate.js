import React from 'react';

const RealDate = Date;

/**
 * Storybook decorator for mocking current time.
 *
 * Sets current time to a fixed value if systemTime is defined in story
 * parameters, otherwise uses the real time.
 */
export const withMockDate = (Story, { parameters }) => {
  if (parameters.systemTime instanceof Date) {
    // eslint-disable-next-line no-global-assign
    Date = class extends Date {
      constructor(...args) {
        if (args.length === 0) {
          super(parameters.systemTime.valueOf());
        } else {
          super(...args);
        }
      }

      static now() {
        return parameters.systemTime.getTime();
      }
    };
  } else {
    // eslint-disable-next-line no-global-assign
    Date = RealDate;
  }

  return <Story />;
};
