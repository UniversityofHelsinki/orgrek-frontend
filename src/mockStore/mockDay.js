import { createSlice } from '@reduxjs/toolkit';
import legacyReducer from '../reducers/dayChangeReducer';

const initialState = {
  selectedDay: new Date(),
};

const daySlice = createSlice({
  name: 'day',
  initialState,
  extraReducers(builder) {
    builder.addDefaultCase(legacyReducer);
  },
});

export const dayReducer = daySlice.reducer;
