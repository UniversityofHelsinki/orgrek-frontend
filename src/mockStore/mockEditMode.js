import { createSlice } from '@reduxjs/toolkit';
import legacyReducer from '../reducers/editModeReducer';

const initialState = {
  edit: false,
};

/**
 * @deprecated used only in the legacy edit view
 */
const editModeSlice = createSlice({
  name: 'editMode',
  initialState,
  extraReducers(builder) {
    builder.addDefaultCase(legacyReducer);
  },
});

/**
 * @deprecated used only in the legacy edit view
 */
export const editModeReducer = editModeSlice.reducer;
