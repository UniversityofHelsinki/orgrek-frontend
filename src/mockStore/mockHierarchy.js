const defaultHierarchies = [
  'tutkimus',
  'henkilosto',
  'toiminnanohjaus',
  'opetus',
  'history',
  'talous',
  'virallinen',
];

export const createHierarchies = (
  hierarchies = null,
  startDate = null,
  endDate = null
) =>
  (hierarchies || defaultHierarchies).map((hierarchy) => ({
    hierarchy,
    startDate,
    endDate,
  }));
