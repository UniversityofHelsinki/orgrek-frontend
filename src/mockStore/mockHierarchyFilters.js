export const hierarchyFilters = [
  {
    id: 6,
    hierarchy: 'talous',
    key: 'type',
    value: 'koontiyksikko',
    startDate: null,
    endDate: null,
  },
  {
    id: 7,
    hierarchy: 'opetus',
    key: 'type',
    value: 'koontiyksikko',
    startDate: null,
    endDate: null,
  },
  {
    id: 12,
    hierarchy: 'opetus',
    key: 'type',
    value: 'tiedekunta',
    startDate: null,
    endDate: null,
  },
  {
    id: 17,
    hierarchy: 'opetus',
    key: 'type',
    value: 'erillinen laitos',
    startDate: null,
    endDate: null,
  },
  {
    id: 21,
    hierarchy: 'talous',
    key: 'type',
    value: 'osasto',
    startDate: '2017-12-31T22:00:00.000+00:00',
    endDate: null,
  },
  {
    id: 22,
    hierarchy: 'opetus',
    key: 'type',
    value: 'osasto',
    startDate: '2017-12-31T22:00:00.000+00:00',
    endDate: null,
  },
  {
    id: 31,
    hierarchy: 'talous',
    key: 'type',
    value: 'tulosyksikko',
    startDate: null,
    endDate: null,
  },
  {
    id: 36,
    hierarchy: 'opetus',
    key: 'type',
    value: 'tutkijakoulu',
    startDate: null,
    endDate: null,
  },
  {
    id: 38,
    hierarchy: 'opetus',
    key: 'type',
    value: 'kandiohjelma',
    startDate: null,
    endDate: null,
  },
  {
    id: 40,
    hierarchy: 'opetus',
    key: 'type',
    value: 'maisteriohjelma',
    startDate: null,
    endDate: null,
  },
  {
    id: 43,
    hierarchy: 'opetus',
    key: 'type',
    value: 'tohtoriohjelma',
    startDate: null,
    endDate: null,
  },
  {
    id: 45,
    hierarchy: 'talous',
    key: 'talous_tunnus',
    value: null,
    startDate: null,
    endDate: null,
  },
  {
    id: 46,
    hierarchy: 'opetus',
    key: 'oppiaine_tunnus',
    value: null,
    startDate: null,
    endDate: null,
  },
  {
    id: 123,
    hierarchy: 'opetus',
    key: 'type',
    value: 'oppiaine',
    startDate: null,
    endDate: null,
  },
  {
    id: 32688,
    hierarchy: 'talous',
    key: 'type',
    value: 'talousvastuullinen yksikko',
    startDate: '2023-03-12T22:00:00.000+00:00',
    endDate: null,
  },
  {
    id: 32272,
    hierarchy: 'opetus',
    key: 'type',
    value: 'erilliset opinnot',
    startDate: '2022-12-31T22:00:00.000+00:00',
    endDate: null,
  },
  {
    id: 32716,
    hierarchy: 'talous',
    key: 'type',
    value: 'yritys_yhteiso',
    startDate: null,
    endDate: null,
  },
  {
    id: 32686,
    hierarchy: 'talous',
    key: 'publicity',
    value: null,
    startDate: null,
    endDate: null,
  },
];
