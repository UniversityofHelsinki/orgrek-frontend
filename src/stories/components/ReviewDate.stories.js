import ReviewDateComponent from '../../components/ReviewDate';
import { withMockStore } from '../../mockStore';

export default {
  component: ReviewDateComponent,
};

export const ReviewDate = {
  decorators: [withMockStore()],
};
