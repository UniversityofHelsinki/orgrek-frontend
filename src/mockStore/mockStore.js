import { configureStore } from '@reduxjs/toolkit';
import { treeReducer } from './mockTree';
import { hierarchyReducer } from './mockHierarchy';
import { createNodeState, nodeReducer } from './mockNode';
import { userReducer } from './mockUser';
import { dayReducer } from './mockDay';
import { nodeDetailsViewReducer } from './mockNodeDetailsView';
import { hierarchyFiltersReducer } from './mockHierarchyFilters';
import { editModeReducer } from './mockEditMode';
import { api, notificationsReducer } from '../store';
import { Provider } from 'react-redux';
import React from 'react';
import { setupListeners } from '@reduxjs/toolkit/query';

export const configureMockStore = (preloadedState = {}) => {
  const store = configureStore({
    preloadedState,
    reducer: {
      [api.reducerPath]: api.reducer,
      tree: treeReducer,
      hr: hierarchyReducer,
      nrd: nodeReducer,
      ur: userReducer,
      dr: dayReducer,
      nvrd: nodeDetailsViewReducer,
      hierarchyFilters: hierarchyFiltersReducer,
      editModeReducer: editModeReducer,
      notifications: notificationsReducer,
    },
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware().concat(api.middleware),
  });

  setupListeners(store.dispatch);

  return store;
};

/**
 * Configures Redux store from initial state and any additional test data
 * passed to this component.
 *
 * Intended to be used as Storybook story decorator, but it should work in
 * tests as well.
 *
 * @param state
 * @param children a Storybook story or any component using the Redux store
 */
export const MockStoreProvider = ({ state, children }) => {
  const store = configureMockStore(state);

  return <Provider store={store}>{children}</Provider>;
};

/**
 * Storybook decorator function which provides the mock Redux store to the story
 *
 * @param state
 * @return a function that can be passed to Storybook decorator array
 */
export const withMockStore = (state) => {
  return function MockStoreDecorator(Story) {
    return (
      <MockStoreProvider state={state}>
        <Story />
      </MockStoreProvider>
    );
  };
};

/**
 * For convenience a decorator that mocks Redux store with the given user.
 *
 * Use this decorator when you need to mock only the current user.
 * Alternatively, withMockStore decorator can mock the whole state.
 *
 * @param user current user
 */
export const withUser = (user) => withMockStore({ ur: { user } });

/**
 * For convenience a decorator that mocks Redux store with the given node data.
 *
 * Use this decorator when you need to mock only the node reducer state.
 * Alternatively, withMockStore decorator can mock the whole state.
 *
 * @param nodeState overrides node reducer initial state
 */
export const withNode = (nodeState) =>
  withMockStore({ nrd: createNodeState(nodeState) });

/**
 * For convenience a decorator that mocks Redux store with the given hierarchy
 * data.
 *
 * Use this decorator when you need to mock only the hierarchy reducer state.
 * Alternatively, withMockStore decorator can mock the whole state.
 *
 * @param hierarchyState overrides hierarchy reducer initial state
 */
export const withHierarchy = (hierarchyState) =>
  withMockStore({ hr: { ...hierarchyState } });
