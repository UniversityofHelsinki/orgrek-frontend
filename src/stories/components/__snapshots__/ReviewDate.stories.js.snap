// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stories/components/ReviewDate ReviewDate smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Ajankohta",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Ajankohta",
      "pressed": undefined,
      "role": "textbox",
      "value": "22.3.2023",
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Choose date, selected date is 22. maalis 2023",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "T채m채 p채iv채",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "stories-components-reviewdate--review-date",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
